meinberlin/apps/votes/admin.py

```diff
@@ -11,19 +12,27 @@ class VotingTokenAdmin(admin.ModelAdmin):
         "token_hash",
         "module",
         "is_active",
-        "package_number",
+        "package",
         "allowed_votes",
     )
-    readonly_fields = ("token", "token_hash", "package_number")
+    readonly_fields = ("token", "token_hash", "package")
     list_display = ("pk", "__str__", "project", "module", "module_name", "is_active")
     list_filter = ("module__project",)
 
+    def delete_model(self, request, obj):
+        obj.package.size -= 1
+        obj.package.save()
+        super().delete_model(request, obj)
+
     def module_name(self, token):
         return token.module.name
 
     def save_model(self, request, obj, form, change):
-        if obj.package_number is None:
-            obj.package_number = VotingToken.next_package_number(obj.module)
+        if obj.package is None:
+            obj.package = TokenPackage.create(module=obj.module, size=1)
+        else:
+            obj.package.size += 1
```

Comment:
It also does that when editing the token, e.g. setting it to inactive.. hm, not sure how to fix this?

Reply:
I just removed it, the package field is read-only anyway, so it will always create a new one